chanbackup/recover.go

```diff
@@ -47,6 +48,14 @@ func Recover(backups []Single, restorer ChannelRestorer,
 			backup.FundingOutpoint)
 
 		err := restorer.RestoreChansFromSingles(backup)
+
+		// If a channel is already present in the channel DB, we can
+		// just continue. No reason to fail a whole set of multi backups
+		// for example. This allows resume of a restore in case another
+		// error happens.
+		if err == channeldb.ErrChanAlreadyExists {
```

Comment:
do we know that this is the only error that we should silence?

Reply:
Yes, restoring channel shells happens in a single DB transaction. If it succeeds, then the second time around, it'll fail when trying to create the channel point bucket: https://github.com/lightningnetwork/lnd/blob/master/channeldb/channel.go#L711.

Reply:
nice, that's what i was looking for